activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter-hal-rest-api/src/main/java/org/activiti/controllers/ProcessDefinitionController.java

```diff
@@ -72,4 +77,69 @@ public ProcessDefinitionResource getProcessDefinition(@PathVariable String id) {
         }
         return resourceAssembler.toResource(processDefinitionConverter.from(processDefinition));
     }
+    
+    @RequestMapping(value = "/{id}/meta", method = RequestMethod.GET)
+    public ProcessDefinitionResource getProcessDefinitionMetadata(@PathVariable String id) {
+    	org.activiti.engine.repository.ProcessDefinition processDefinition = repositoryService.createProcessDefinitionQuery().processDefinitionId(id).singleResult();
+        if (processDefinition == null) {
+            throw new ActivitiException("Unable to find process definition for the given id:'" + id + "'");
+        }
+        Process process = repositoryService.getBpmnModel(id).getMainProcess();
+        List<String []> variables = new ArrayList<String[]>();
+        if(!process.getExtensionElements().isEmpty())
+        {
+        	Iterator<List<ExtensionElement>> it = process.getExtensionElements().values().iterator();
+        	while(it.hasNext())
+        	{
+        		List<ExtensionElement> extensionElementList = it.next();
+        		Iterator<ExtensionElement> it2 = extensionElementList.iterator();
+        		while(it2.hasNext())
+        		{
+        			ExtensionElement ee = it2.next();
+        			String[] variable = new String[2];
+        			variable[0] = ee.getAttributeValue(ee.getNamespace(), "variableName");
+        			variable[1] = ee.getAttributeValue(ee.getNamespace(), "variableType");
+        			variables.add(variable);
+        		}
+        	}
+        }
+        List<String> users = new ArrayList<String>();
+        List<String> groups = new ArrayList<String>();
+        List<String []> userTasks = new ArrayList<String []>();
+        List<String []> serviceTasks = new ArrayList<String []>();
+        List<FlowElement> flowElementList = (List<FlowElement>) process.getFlowElements();
+        for (FlowElement f : flowElementList) {
+        	if(f.getClass().equals(UserTask.class))
+        	{
+        		UserTask userTask = (UserTask) f;
+        		String[] t = new String[2];
+        		t[0] = userTask.getName();
+        		t[1] = userTask.getDocumentation();
+        		userTasks.add(t);
+        		users.addAll(userTask.getCandidateUsers());
+        		groups.addAll(userTask.getCandidateGroups());
+        	}
+        	if(f.getClass().equals(ServiceTask.class))
+        	{
+        		ServiceTask serviceTask = (ServiceTask) f;
+        		String[] s = new String[2];
+        		s[0] = serviceTask.getName();
+        		s[1] = serviceTask.getImplementation();
+        		serviceTasks.add(s);
+        	}
+        }
+        
+        ProcessDefinition pd = new ProcessDefinition(
```

Comment:
@erdemedeiros I think that we have a converter to do these transformations from ProcessDefinition (engine) to our ProcessDefinition. @gmalanga I would recommend you to use that converter to make sure that we cover all the properties required. Also if there is something missing in those converters that need to be updated.